.github/workflows/main.yml

```diff
@@ -24,6 +24,10 @@ on:
   pull_request:
     types: [opened, synchronize, reopened]
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref || github.run_id }}
+  cancel-in-progress: true
```

Comment:
What does this do? It's not totally clear to me from the GitHub documentation.

Reply:
Thanks for looking into this, will make the suggested change.